src/node_sqlite.cc

```diff
@@ -1897,6 +1906,72 @@ void StatementSync::Run(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(result);
 }
 
+void StatementSync::Columns(const FunctionCallbackInfo<Value>& args) {
+  StatementSync* stmt;
+  ASSIGN_OR_RETURN_UNWRAP(&stmt, args.This());
+  Environment* env = Environment::GetCurrent(args);
+  THROW_AND_RETURN_ON_BAD_STATE(
+      env, stmt->IsFinalized(), "statement has been finalized");
+  int num_cols = sqlite3_column_count(stmt->statement_);
+  Isolate* isolate = env->isolate();
+  LocalVector<Value> cols(isolate);
+  LocalVector<Name> col_keys(isolate,
+                             {FIXED_ONE_BYTE_STRING(isolate, "column"),
+                              FIXED_ONE_BYTE_STRING(isolate, "database"),
+                              FIXED_ONE_BYTE_STRING(isolate, "name"),
+                              FIXED_ONE_BYTE_STRING(isolate, "table"),
+                              FIXED_ONE_BYTE_STRING(isolate, "type")});
+
+  cols.reserve(num_cols);
+  for (int i = 0; i < num_cols; ++i) {
+    LocalVector<Value> col_values(isolate);
```

Comment:
This can be moved to after the first if statement. Since it would be unnecessary to create this local vector if that if statement early terminates.

Reply:
I would prefer not to make that change because I feel that it makes the code less readable in order to make a very small optimization for an edge case that should almost never happen.